plugins/bazaar/src/api.ts

```diff
@@ -148,12 +148,15 @@ export class BazaarClient implements BazaarApi {
     );
   }
 
-  async getProjects(): Promise<any> {
+  async getProjects(limit?: number, order?: string): Promise<any> {
     const baseUrl = await this.discoveryApi.getBaseUrl('bazaar');
+    const url =
+      order && limit
+        ? `projects/${encodeURIComponent(limit)}/${encodeURIComponent(order)}`
+        : `/projects`;
 
-    return await this.fetchApi
-      .fetch(`${baseUrl}/projects`)
-      .then(resp => resp.json());
+    const data = await this.fetchApi.fetch(`${baseUrl}/${url}`);
```

Comment:
a hint here - the backend uses `router.use(errorHandler())` which captures backend errors in a nice, rich, detailed way. To make the client leverage this, you can add a line right here which does

```ts
if (!data.ok) {
  throw await ResponseError.fromResponse(data);
}
```

(where `ResponseError` is in the `@backstage/errors` package)

This ensures that any error details in the response body are captured and available to the caller. Especially the `<ErrorPanel>` component, which lets the user see all of those error details fully and better understanding what went wrong exactly where.